### ISSUE TITLE
/predict/neural can serve predictions from randomly initialized weights

### ISSUE THREAD

**Issue by Evgeniy2001Poluhin** [Evgeniy2001Poluhin/-SORA.Earth#320]
Найдено при инвентаризации #191 на `main@1a68015` (весь CI зелёный). К #191 дефект
не относится и не должен чиниться внутри него: #191 — про хранение артефактов,
а здесь неверный результат уходит наружу.

## Что измерено

```
models/pytorch_mlp.pth        нет в git (git ls-files — пусто), нет на диске
.dockerignore:50              models/ исключён из образа
app/main.py:502               nn_model = SoraNet()        — безусловно
app/main.py:503-506           load_state_dict(...)        — только if os.path.exists(NN_PATH)
```

`SoraNet()` конструируется всегда. Поэтому `nn_model is not None` истинно
независимо от того, загрузились веса или нет, и при отсутствии файла сеть
остаётся со **случайной инициализацией** — `torch` заполняет слои по умолчанию,
`eval()` на это не влияет.

Проверка `os.path.exists` здесь не защита, а именно то, что превращает
отсутствие артефакта в тихую подмену: без неё был бы `FileNotFoundError` при
старте.

## Кто это потребляет

Четыре маршрута вызывают `_nn_forward(nn_model, ...)` без единой проверки, что
веса загружены:

| маршрут | вызов |
|---|---|
| `POST /api/v1/predict/neural` | `app/api/predict.py:89` → `:101` |
| `POST /api/v1/predict/stacking` | `app/api/predict.py:120` → `:137` |
| `POST /api/v1/predict/compare` | `app/api/predict.py:163` → `:174` |
| `POST /api/v1/report/pdf` | `app/api/evaluate.py:457` → `:481` |

Последний кладёт число в PDF-отчёт, который получает пользователь.

## Почему это не видно

Четыре индикатора докладывают, что модель загружена, и все четыре опираются на
`nn_model is not None`, то есть на выражение, которое не может быть ложным:

| индикатор | код |
|---|---|
| `GET /api/v1/health` | `app/api/system.py:14` через `_check_models()` (`:11-16`), `:56` |
| `GET /api/v1/ready` (readiness probe) | `app/api/system.py:79-81` — 503 только при `status != "healthy"` |
| `GET /api/v1/analytics/metrics/model-health` | `app/api/analytics.py:249` — `"pytorch_mlp": {"loaded": nn_model is not None}` |
| `GET /api/v1/analytics/summary` | `app/api/analytics.py:460`, `:527` |
| `GET /system/health` | `app/main.py:835` — `"ml_models": "ok" if rf_model and xgb_model and nn_model else "warn"` |

Readiness probe существует ровно для того, чтобы неисправный экземпляр не
попал под трафик. Этот отказ он обнаружить не может.

`CLAUDE.md` в разделе Models называет `models/pytorch_mlp.pth` одной из четырёх
моделей, загружаемых при старте. Файла в репозитории нет.

## Состояние production не проверено

Файла нет в git, а `.dockerignore:50` исключает `models/` из образа, поэтому в
контейнер каталог попадает только bind-монтированием хостового чекаута
(`docker-compose.prod.yml:109,193`). Отсюда следует, что production располагает
файлом только если его положили на хост вне git. Это **вывод, а не измерение** —
требуется отдельная read-only проверка на хосте, и до неё утверждать, что
production отдаёт случайные веса, нельзя.

В любом свежем клоне и в любой сборке из образа дефект воспроизводится.

## Ожидаемое поведение

1. Модель считается загруженной только после успешной загрузки весов. Признак
   загруженности не должен выводиться из того, что объект сконструирован.
2. При отсутствии или непрочитанных весах `/predict/neural` возвращает
   стабильный отказ с безопасным телом, а не предсказание. Тело не раскрывает
   файловых путей — как это уже сделано в `ModelSource` (`app/model_source.py:113`,
   тест `tests/test_loader_uses_resolver.py:177`).
3. `/predict/stacking`, `/predict/compare` и `/report/pdf` либо отказывают, либо
   считают результат без нейросетевой компоненты, явно помечая это в ответе.
   Молча подставлять число из необученной сети нельзя.
4. `/health`, `/ready` и оба analytics-маршрута отражают фактическое состояние
   весов. `/ready` обязан отвечать 503, если серверная конфигурация требует
   нейросеть, а весов нет.
5. Решить и записать, обязательна ли нейросеть: если да — отсутствие весов
   должно останавливать старт; если нет — она деградирует явно, а не молча.
   Сегодня не выбрано ни то, ни другое.

## Регрессионные тесты

- Отсутствие `pytorch_mlp.pth` → `/predict/neural` не возвращает 200 с числом.
- Отсутствие весов → `loaded` ложно во всех четырёх индикаторах; `/ready`
  отвечает 503, если сеть объявлена обязательной.
- **Негативный контроль:** при наличии валидных весов те же тесты проходят по
  успешной ветке. Без него проверка отсутствия ничего не доказывает.
- Присутствие весов → предсказание детерминировано для фиксированного входа
  (два вызова дают одно значение), что отличает загруженную сеть от заново
  инициализированной.
- Тест, доказывающий, что признак загруженности не выводится из
  `nn_model is not None`: подмена объекта на сконструированный без весов должна
  давать `loaded == false`.

## Критерии готовности

- Ни один из четырёх маршрутов не отдаёт результат, посчитанный необученной сетью.
- Ни один индикатор не докладывает `loaded: true` при незагруженных весах.
- Решение об обязательности нейросети принято и зафиксировано в `CLAUDE.md`,
  раздел Models приведён в соответствие с содержимым репозитория.
- Регрессионные тесты выше зелёные, включая негативный контроль.
- Судьба самого артефакта решена отдельно: либо он добавляется в seed через LFS
  (тогда `.github/workflows/ci.yml:61-90` должен его проверять — сейчас список
  покрывает 5 артефактов из 10), либо маршруты официально работают без него.

## Приоритет

P0/P1. Затрагивает корректность внешнего результата, а не внутреннюю чистоту:
четыре публичных маршрута, включая PDF-отчёт, и readiness probe, который не
может об этом сообщить.

Связано: #191 (хранение артефактов; этот дефект — следствие того, что наличие
артефакта зависит от монтирования, которое никто не проверяет).
